tests/test_cli.py

```diff
@@ -116,3 +102,40 @@ def test_cli_errors(args: list[str], expected_output: str, script_runner: Script
     ret = script_runner.run(["mqt.bench.cli", *args])
     assert not ret.success
     assert expected_output in ret.stderr
+
+
+def test_cli_output_formats_and_save(tmp_path: Path, script_runner: ScriptRunner) -> None:
+    """Test output formats and save functionality."""
+    # Test QASM3 prints to stdout when not saving
+    ret_qasm3 = script_runner.run([
+        "mqt.bench.cli",
+        "--level", "alg",
+        "--algorithm", "ghz",
+        "--num-qubits", "5",
+        "--output-format", "qasm3",
+    ])
+    assert ret_qasm3.success
+    # QASM3 header should appear and not relate to save
+    assert "MQT Bench version:" in ret_qasm3.stdout
+    assert "OPENQASM" in ret_qasm3.stdout
+
+    # Test saving QPY to a file
+    target_dir = str(tmp_path)
+    target_file = "mybench"
+    ret_qpy = script_runner.run([
+        "mqt.bench.cli",
+        "--level", "alg",
+        "--algorithm", "ghz",
+        "--num-qubits", "5",
+        "--output-format", "qpy",
+        "--save",
+        "--target-directory", target_dir,
+        "--target-filename", target_file,
+    ])
+    # File should exist
+    assert ret_qpy.success
+    # Should output the path to the saved file
+    expected_path = str(Path(target_dir) / f"{target_file}.qpy")
+    assert expected_path in ret_qpy.stdout
+    # File should exist
+    assert (tmp_path / f"{target_file}.qpy").is_file()
```

Comment:
Might be better as individual smaller tests to kind of keep the spirit of a "unit" test.